test/neurx/optimization_functions/optimizers_test.exs

```diff
@@ -48,11 +48,9 @@ defmodule Neurx.OptimizersTest do
     assert(connection_pid)
     connection = Connection.get(connection_pid)
     assert(connection)
-    assert(connection.weight == 0.4)
```

Comment:
Can we assert that the Connection.weight is >0 and <1?

This would be affected in multiple places...

Reply:
Good call.  I will add said asserts in optimizers_test and connection_test